src/zrc/api/validators.py

```diff
@@ -204,3 +213,126 @@ class ZaakBesluitValidator:
     def __call__(self, zaak: Zaak):
         if zaak.zaakbesluit_set.exists():
             raise serializers.ValidationError(self.message, code=self.code)
+
+
+class EitherFieldRequiredValidator:
+    default_message = _("One of %(fields)s must be provided")
+    default_code = "invalid"
+
+    def __init__(self, fields: Iterable[str], message: str = "", code: str = ""):
+        self.fields = fields
+        self.message = (message or self.default_message) % {"fields": ", ".join(fields)}
+        self.code = code or self.default_code
+
+    def __call__(self, attrs: dict):
+        values = [attrs.get(field, None) for field in self.fields]
+        if not any(values):
+            raise serializers.ValidationError(self.message, code=self.code)
+
+
+class JQExpressionValidator:
+    message = _("This is not a valid jq expression.")
+    code = "invalid"
+
+    def __call__(self, value: str):
+        try:
+            jq.compile(value)
+        except ValueError:
+            raise serializers.ValidationError(self.message, code=self.code)
+
+
+class ObjectTypeOverigeDefinitieValidator:
+    code = "invalid"
+
+    def __call__(self, attrs: dict):
+        object_type_overige_definitie: Optional[dict] = attrs.get(
+            "object_type_overige_definitie"
+        )
+        object_url: str = attrs.get("object", "")
+
+        if not object_type_overige_definitie:
+            return
+
+        # object type overige definitie can only be used with object URL reference
+        if object_type_overige_definitie and not object_url:
+            raise serializers.ValidationError(
+                {
+                    "object_url": _(
+                        "Indien je `objectTypeOverigeDefinitie` gebruikt, dan moet je "
+                        "een OBJECT url opgeven."
+                    )
+                },
+                code="required",
+            )
```

Comment:
Can we discuss this restriction?

I thought that if zaakobject.object is provided that probably means that you deal with Objects API (or other known API) and therefore you can retrieve the related object_type and inspect its JSON schema without zaakobject.object_type_other_definition field.
And zaakobject.object_type_other_definition is used for data which don't have API to access it, i.e. for zaakobject.objectIdentifier.

So I thought it should be the opposite: to use zaakobject.object_type_other_definition only with zaakobject.objectIdentifier, since if you have the API to store the objects you probably already knows its shape.